### PR TITLE
ROOT master - integrate with TBB 2021

### PR DESCRIPTION
**Pull request by mrodozov**

### CODE REVIEW

**Comment by mrodozov** [comment]
please test with https://github.com/cms-sw/cmsdist/pull/6544

**Comment by mrodozov** [comment]
please test with #6544, https://github.com/cms-externals/heppdt/pull/1

**Comment by mrodozov** [comment]
please test with #6544, cms-externals/heppdt#1

**Comment by mrodozov** [comment]
please test with #6544, cms-externals/heppdt#1

**Comment by mrodozov** [comment]
I think we had something that was installing the TBBCmake for DD4Hep (we had that dd4hep issue some time ago) :/

**Comment by mrodozov** [comment]
please test

**Comment by mrodozov** [comment]
abort

**Comment by mrodozov** [comment]
please test with #6544, cms-externals/heppdt#1

**Comment by mrodozov** [comment]
oh well now dd4hep cmake is fine, but it fails because of 
tbb::task_scheduler_init::default_num_threads();

**Comment by mrodozov** [comment]
checking cupla and alpaka before running the test again

File: dd4hep.spec [hunk 1]
```diff
@@ -23,8 +23,7 @@ CMAKE_ARGS="-DCMAKE_INSTALL_PREFIX='%{i}' \
       -DDD4HEP_USE_PYROOT=ON \
       -DCMAKE_CXX_STANDARD=17 \
       -DCMAKE_BUILD_TYPE=Release \
-      -DDD4HEP_USE_TBB=ON \
-      -DCMAKE_PREFIX_PATH=${TBB_ROOT}/cmake/TBB;${CLHEP_ROOT};${XERCES_C_ROOT}"
+      -DCMAKE_PREFIX_PATH=${TBB_ROOT}/lib/cmake/TBB;${CLHEP_ROOT};${XERCES_C_ROOT}"
```
Comment (smuzaffar) [inline dd4hep.spec]: @mrodozov , do we still need `${TBB_ROOT}/lib/cmake/TBB` and `tbb` deps here?

Comment (mrodozov) [reply dd4hep.spec]: right. we don't. it's configuring without it (I checked)

**Comment by mrodozov** [comment]
please test with #6544, cms-externals/heppdt#1
heppdt#1 was merged but not changed in cmsdist, we still need it. alpaka and cupla built with new tbb

**Comment by mrodozov** [comment]
:/ 
`/cvmfs/cms-ib.cern.ch/nweek-02664/slc7_amd64_gcc900/external/gcc/9.3.0/bin/../lib/gcc/x86_64-unknown-linux-gnu/9.3.0/../../../../x86_64-unknown-linux-gnu/bin/ld: /cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc900/cms/cmssw/CMSSW_11_3_ROOT6_X_2021-01-19-2300/lib/slc7_amd64_gcc900/libFWCorePluginManager.so: undefined reference to tbb::internal::concurrent_vector_base_v3::internal_capacity() const'`
it appears we'll have to adjust cmssw as well

**Comment by makortel** [comment]
> it appears we'll have to adjust cmssw as well

It looks to me that we need to wait for @Dr15Jones to finish the migration from `tbb::task` to `tbb::task_group`.

**Comment by mrodozov** [comment]
please test with #6544

**Comment by Dr15Jones** [comment]
The tbb::task API was deprecated and may have been fully removed in the newest version of oneTBB. I'm in the process of removing CMSSW's useage of tbb::task but it is still very much a work in progress.

**Comment by mrodozov** [comment]
no worries,
I'm testing weekly ROOT changes in both 622 and master for integration purposes.
We can survive some time without integrating lastest ROOTs :)

**Comment by mrodozov** [comment]
closing this one we'll use #6612 for DD4 testing

**Comment by mrodozov** [comment]
see https://github.com/cms-sw/cmsdist/pull/6622 root master (externals) integration.

**Pull request closed without merging**